docker/convert.go

```diff
@@ -114,7 +117,7 @@ func ports(c *config.ServiceConfig) (map[nat.Port]struct{}, nat.PortMap, error)
 }
 
 // Convert converts a service configuration to an docker API structures (Config and HostConfig)
-func Convert(c *config.ServiceConfig, ctx project.Context) (*container.Config, *container.HostConfig, error) {
+func Convert(c *config.ServiceConfig, ctx project.Context, clientFactory composeclient.Factory) (*container.Config, *container.HostConfig, error) {
```

Comment:
Is there any way we can get around having to use the Docker daemon here? The process of converting a `ServiceConfig` to a container `Config` and `HostConfig` is really useful even in use cases without a Docker daemon. We're pretty dependent on this for Rancher Compose.

If not, maybe we can just split this into two functions, one that doesn't depend on Docker and another that does (while reusing the former).

I was actually thinking about moving the conversion process to a separate package to make it explicitly independent of the `docker` package (which I see as more responsible for the actual interactions with the Docker daemon).